### PR TITLE
Refactored option handling

### LINKED ISSUE #11: evaluate tool error
There is a bug in the `./evaluateTool` command:
```
> ./evaluateTool -t1 -mis0 -mil0 -mip0 -mit0
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at tasks.EvaluateTool.main(EvaluateTool.java:243)
```

This is caused by the CLI parser, thinking `-m` was a matcher. It tries to split the matcher string and fails.

Suggested solution:
- Put a error message when the matcher string cannot be parsed
- use two long option names for those min/max options instead of a short and a long option, so the call above would become `./evaluateTool -t1 --mis0 --mil0 --mip0 --mit0`

I'd be happy to do a PR for this, if this is wanted.

While browsing the code, I was thinking of [picocli](https://picocli.info/quick-guide.html) which would make the whole option parsing more compact. If this is an option, I could port the whole project from apache cli to picocli as part of the PR, if you are interested.

### PR DESCRIPTION
**Pull request by qw3ry**
Closes #11 

It has the following changes:

- Changed option parsing library to picocli
  - Renamed some options during this process. Multi-character short options (like `-mil`) became long options (`--mil`). Some got an additional short option.
  - The default help option of picocli also adds a `--version` option. I added a `Version.java` that is generated in the build process and includes the git commit hash.
- Added a description for each command (taken from the readme) that is shown in the usage help
- Added `./bcb` as "uber" command that can call all the other commands. `./bcb` also provides a detailed help on each subcommand.
- Updated the readme
  - To markdown format (better rendering on github)
  - To reflect the changed options

### CODE REVIEW

File: src/util/Version.java.template [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package util;
+
+import picocli.CommandLine;
+
+public class Version implements CommandLine.IVersionProvider {
+    public static final String version = "%%VERSION%%";
+
+    @Override
+    public String[] getVersion() {
+        return new String[] {"1.0.0." + version};
```
Comment (jeffsvajlenko) [inline src/util/Version.java.template]: I've made a release for before these changes to be 1.0.  Could make this 1.1 or 2.0 to reflect updated interface.

Comment (jeffsvajlenko) [inline src/util/Version.java.template]:
```suggestion
        return new String[] {"1.1.0." + version};
```

**Review by jeffsvajlenko — APPROVED** [review]
Looks good.  I created an initial release before these changes due to lots of edits (in case of new bugs).  Lets make this the new latest release (1.1?).

File: src/util/Version.java.template (hunk 1 above)
Comment (qw3ry) [reply src/util/Version.java.template]: Sure ;)

Comment (qw3ry) [reply src/util/Version.java.template]: Sure

**Comment by jeffsvajlenko** [comment]
Hi @qw3ry.  Feel free to complete this PR if you have found no issues running locally.  Otherwise, once I have time to manually test the changes locally, I will integrate it.  Thanks!

**Comment by qw3ry** [comment]
I don't think I've permission to do so.
It runs fine for me. But feel free to wait until you tested it yourself

**Comment by jeffsvajlenko** [comment]
Introducing these changes as part of https://github.com/jeffsvajlenko/BigCloneEval/pull/21.

**Pull request merged**